cli/lib/commands/init.js

```diff
@@ -0,0 +1,210 @@
+  createToml,
+  fileExists,
+  isValidSemver,
+  writeFile
+} = require('../utils');
+
+const REJECTION_MSGS = {
+  REQUIRED: 'This is required',
+  validateName: "Sorry, that's not a valid name",
+  validateVersion: "Sorry, that's not a valid semver version"
+};
+
+const VALID_YES_NO = {
+  y: 1,
+  yes: 1,
+  n: 0,
+  no: 0
+};
+
+const QUESTIONS = {
+  TOML_EXISTS: {
+    MESSAGE: 'Package.toml already exists. Continue (Y/N)?: ',
+    REQUIRED: true
+  },
+  NAME: {
+    MESSAGE: 'Name: ',
+    REQUIRED: true
+  },
+  VERSION: {
+    MESSAGE: 'Version: ',
+    REQUIRED: true
+  },
+  ENTRY: {
+    MESSAGE: 'Entry: ',
+    REQUIRED: false
+  },
+  TYPE: {
+    MESSAGE: 'Type :',
+    REQUIRED: false
+  },
+  LICENSE: {
+    MESSAGE: 'License: ',
+    REQUIRED: false
+  },
+  DESCRIPTION: {
+    MESSAGE: 'Description: ',
+    REQUIRED: false
+  },
+  HOMEPAGE: {
+    MESSAGE: 'Homepage: ',
+    REQUIRED: false
+  },
+  AUTHOR: {
+    MESSAGE: 'Author: ',
+    REQUIRED: false
+  },
+  REPOSITORY: {
+    MESSAGE: 'Repository: ',
+    REQUIRED: false
+  }
+};
+
+/**
+ * Returns string for the path to write Package.toml
+ */
+function tomlLocation() {
+  return `${process.cwd()}/Package.toml`;
+}
+
+/**
+ * Returns boolean based upon whether a user's answer is in the VALID_YES_NO hash
+ */
+function validateYesNo(ans) {
+  if (!ans) {
+    return false;
+  }
+
+  return VALID_YES_NO[ans] !== undefined;
+}
+
+/**
+ * Validates the name provided by the user
+ */
+function validateName(name) {
+  // TODO: Uese Regex for "name@registry-domain.dev/package_name"?
+  return name && name.length > 0;
+}
+
+/**
+ * Validates the version provided by the user
+ */
+function validateVersion(version) {
+  return isValidSemver(version);
+}
+
+/**
+ * Used to provide consistent casing for hash lookups
+ */
+function lowercase(str) {
+  return str.toLowerCase();
+}
+
+/**
+ * Minimal prompt for asking users for input
+ */
+function ask(question, rl) {
+  return new Promise((resolve, reject) => {
+    rl.question(question, answer => resolve(answer));
+  }).catch(e => {
+    console.error(e);
+  });
+}
+
+/**
+ * Asks user a question until a valid answer is given and applies any
+ * desired transformations to the value returned if provided
+ */
+async function askQuestion(question, rl, validator, transform = undefined) {
+  let invalid = true;
+  let ans = undefined;
+  const { MESSAGE, REQUIRED } = question;
+
+  while (invalid) {
+    ans = await ask(MESSAGE, rl);
+
+    if (transform) {
+      ans = transform(ans);
+    }
+
+    if (ans.length < 1 && REQUIRED) {
+      console.error(REJECTION_MSGS.REQUIRED);
+      continue;
+    }
+
+    if (!validator || validator(ans)) {
+      invalid = false;
+    } else {
+      console.error(REJECTION_MSGS[validator.name]);
+    }
+  }
+
+  return ans;
+}
+
+/**
+ * Exported function used as `ds create`
+ */
+async function create(opts) {
+  await writeFile(`${process.cwd()}/package.json`, createPackageJson());
```

Comment:
For consistency, I'd either get rid of `tomlLocation` or create an additional helper for `jsonLocation`.